src/django_components/component.py

```diff
@@ -570,6 +572,19 @@ def on_render_after(self, context: Context, template: Template, content: str) ->
         """
         pass
 
+    # #####################################
+    # EXTENSIONS
+    # #####################################
+
+    # NOTE: These are the classes and instances added by defaults extensions. These fields
+    # are actually set at runtime, and so here they are only marked for typing.
+    Defaults: Type[ComponentDefaults]
+    defaults: ComponentDefaults
+    View: Type[ComponentView]
+    view: ComponentView
+    Url: Type[ComponentUrl]
+    url: ComponentUrl
```

Comment:
I've added these so that the attributes / classes of the built-in extensions are recognized by type checkers.

Ideally the typing for the Python class could be dynamically extended with extra attributes defined by third parties, similar how interfaces can be extended in TypeScript.

But Python's typing system isn't that advanced. So the next best thing I can do is to explicitly type the known extensions here.